pkg/terra/relay.go

```diff
@@ -97,62 +88,39 @@ func (r *Relayer) Healthy() error {
 	return r.chainSet.Healthy()
 }
 
-// NewOCR2Provider creates a new OCR2ProviderCtx instance.
-func (r *Relayer) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relaytypes.OCR2ProviderCtx, error) {
-	spec, ok := s.(OCR2Spec)
-	if !ok {
-		return nil, errors.New("unsuccessful cast to 'terra.OCR2Spec'")
-	}
-
-	chain, err := r.chainSet.Chain(r.ctx, spec.ChainID)
-	if err != nil {
-		return nil, err
-	}
-	chainReader, err := chain.Reader(spec.NodeName)
-	if err != nil {
-		return nil, err
-	}
-	msgEnqueuer := chain.TxManager()
+func (r *Relayer) NewConfigWatcher(args relaytypes.ConfigWatcherArgs) (relaytypes.ConfigWatcher, error) {
+	return newConfigWatcher(r.ctx, r.lggr, r.chainSet, args)
```

Comment:
Might be worth being defensive here by never returning the typed nil `(*configWatcher)(nil)`:
```suggestion
	configWatcher := newConfigWatcher(r.ctx, r.lggr, r.chainSet, args)
	if err != nil {
		return nil, err
	}
	return configWatcher, nil
```